stable/opa/templates/deployment.yaml

```diff
@@ -15,26 +15,88 @@ spec:
         app: {{ template "opa.fullname" . }}
       name: {{ template "opa.fullname" . }}
     spec:
+{{- if .Values.authz.enabled }}
+      initContainers:
+        - name: initpolicy
+          image: {{ .Values.mgmt.image }}:{{ .Values.mgmt.imageTag }}
+          imagePullPolicy: {{ .Values.mgmt.imagePullPolicy }}
+          command:
+          - /bin/sh
+          - -c
+          - |
+            tr -dc 'A-F0-9' < /dev/urandom | dd bs=1 count=32 2>/dev/null > /authz/mgmt-token
+            TOKEN=`cat /authz/mgmt-token`
+            cat > /authz/authz.rego <<EOF
+            package system.authz
+            default allow = false
+            allow { input.path = [""]; input.method = "POST" }
+            allow { input.path = [""]; input.method = "GET" }
+            allow { input.identity = "$TOKEN" }
+            EOF
+          volumeMounts:
+            - name: authz
+              mountPath: /authz
+{{- end }}
       containers:
         - name: opa
           image: {{ .Values.image }}:{{ .Values.imageTag }}
+          imagePullPolicy: {{ .Values.imagePullPolicy }}
           args:
             - "run"
             - "--server"
             - "--config-file=/config/config.yaml"
             - "--tls-cert-file=/certs/tls.crt"
             - "--tls-private-key-file=/certs/tls.key"
             - "--addr=0.0.0.0:443"
+{{- if .Values.authz.enabled }}
+            - "--authentication=token"
+            - "--authorization=basic"
+            - "/authz/authz.rego"
+            - "--ignore=.*"
+{{- end }}
+{{- if .Values.mgmt.enabled }}
+            - "--insecure-addr=127.0.0.1:8181"
+{{- end }}
           volumeMounts:
             - name: certs
               readOnly: true
               mountPath: /certs
             - name: config
               readOnly: true
               mountPath: /config
+            - name: authz
```

Comment:
This needs to be wrapped with the authz enabled check otherwise the upgrade command will fail.

Reply:
Your right. Fixed.

Almost tempted to just untemplate out all the emptydir /authz and just have it always enabled but unused to simplify the templates.